developer/src/kmc-convert/src/keylayout-to-kmn/keylayout-to-kmn-converter.ts

```diff
@@ -1,27 +1,990 @@
+              /* e.g. [ [ 'anyOption', 'Caps' ] ] */      const b2_prev_deadkeyModifier_arr: string[] = this.get_Modifier_array__From__KeyModifier_array(data_ukelele.arrayOf_Modifiers, b2_prev_deadkey_arr);
+              // ...........................................................................................................................................................................................
+              // Data of Block Nr 6 ........................................................................................................................................................................
+              // create an array[action id,state,output] from all state-output-pairs that use state = b5_value_next (e.g. use 1 in  <when state="1" output="â"/> ) .........................................
+              /*  eg:[ [ 'a9','1','â'] ]*/                const b6_actionId_arr: string[][] = this.get_ActionStateOutput_array__From__ActionState(jsonObj, b5_value_next);
+              // ...........................................................................................................................................................................................
+
+              // Data of Block Nr 1  .......................................................................................................................................................................
+              // create array[Keycode,Keyname,action id,actionIndex,output] and array[Keyname,action id,behaviour,modifier,output] .........................................................................
+              /*  eg: ['49','K_SPACE','a0','0','Â'] */    const b1_keycode_arr: string[][] = this.get_KeyActionOutput_array__From__ActionStateOutput_array(jsonObj, b6_actionId_arr);
+              /*  eg: ['K_SPACE','a0','0','NCAPS','Â'] */ const b1_modifierKey_arr: string[][] = this.get_KeyMBehaviourModOutputArray__from__KeyActionBehaviourOutput_array(jsonObj, b1_keycode_arr, isCapsused);
+                // ...........................................................................................................................................................................................
+
+                for (let n1 = 0; n1 < b2_prev_deadkeyModifier_arr.length; n1++) {
+                  for (let n2 = 0; n2 < b2_prev_deadkeyModifier_arr[n1].length; n2++) {
+                    for (let n3 = 0; n3 < b2_prev_deadkey_arr.length; n3++) {
+                      for (let n4 = 0; n4 < b4_deadkeyModifier_arr.length; n4++) {
+                        for (let n5 = 0; n5 < b4_deadkeyModifier_arr[n4].length; n5++) {
+                          for (let n6 = 0; n6 < b4_deadkey_arr.length; n6++) {
+                            for (let n7 = 0; n7 < b1_modifierKey_arr.length; n7++) {
```

Comment:
😨 

I imagine the human-written KMN equivalent would use stores here?  The smell I'm picking up is that the .keylayout's definitions correspond to something that corresponds to the role of KMN keyboard stores.  (Granted, stores don't handle modifier sets on input keys, so some loops would still be necessary for that aspect.)

It looks as if these loops are explicitly writing out rules for a full set of permutations across multiple stores - multiple sets of values that can vary completely independently from each other.

It'd be nice to actually _convert_ the pattern to stores if possible.  That would certainly yield a more human-readable result.  That said, I'm not the expert in .keylayout here, and it's a "would be nice", not a necessity for our initial implementation.